--- util.py.orig	2016-09-06 14:08:43.365668015 +0000
+++ util.py	2016-09-06 14:09:23.537659594 +0000
@@ -192,7 +192,7 @@
     if not log_file:
         myHandler = logging.StreamHandler(sys.stdout)
     else:
-        myHandler = logging.FileHandler(log_file)
+        myHandler = logging.WatchedFileHandler(log_file)
 #    consoleHandler = logging.StreamHandler(sys.stderr)
     myHandler.setFormatter(formatter)
     myHandler.setLevel(logging.DEBUG)
